src/lib/statistics/generate_pruning_statistics.hpp

```diff
@@ -8,6 +8,11 @@ namespace hyrise {
 class Chunk;
 class Table;
 
+/**
+ * Check whether we need to generate pruning statistics for a chunk.
+ */
+bool pruning_statistics_should_be_generated_for_chunk(const std::shared_ptr<Chunk>& chunk);
```

Comment:
I don't like the naming.  `chunk_pruning_statistics_can_be_added` sucks as well. Can we find something better?

Reply:
`finalized_chunk_lacks_(pruning_)statistics`? Or negate the condition and call it `finalized_chunk_has_(pruning_)statistics`?

Reply:
Problem with `immutable_chunk_lacks_pruning_statistics` (switched to immutable) is that it is not really clear what the result for an mutable chunk should be.
"chunk qualifies for getting pruning stat" ish?

But we also shouldn't overthink this here. Hmpf!

Reply:
Can we conduct a representative survey?

But what's against `immutable_chunk_has_pruning_statistics`? Or `chunk_(is)_immutable_and_has_pruning_statistics` (quite long though). To me the first one sounds okay, but I'd also say let's not make this a longer discussion than necessary.

Reply:
`immutable_chunk_has_pruning_statistics` sounds strange as it is not clear what the result is when we pass a mutable chunk. Sure, with knowledge one knows that those chunks do not have any statistics, but I am curious if we can find a really accurate name. :(

> Or chunk_(is)_immutable_and_has_pruning_statistics (quite long though) [...]

`chunk_is_mutable_or_has_pruning_statistics`? Dann ist klar, dass man skippen kann. Ist ja kein wirkliches "and".

Reply:
But yes, I'm five with your suggestion

Reply:
Why are you talking in German?

Reply:
Not sure if I really like `chunk_is_mutable_or_has_pruning_statistics`, but if we can agree on it, let's just take it?

My problem with `chunk_(is)_immutable_and_has_pruning_statistics` is, dass ich das ja gar nicht nutzen kann, oder? Wenn er immutable ist ohne Stats (return false), möchte ich die hinzufügen. Wenn er mutable ist (return false), kann ich nichts machen. Beide male ist das Ergebnis aber halt false.